templates/mixplank/next/public/script.js

```diff
@@ -0,0 +1,193 @@
+        const attributesToMask = [
+            'username', 'user', 'user_id', 'userid',
+            'password', 'pass', 'pin', 'passcode', "token", "api_token",
+            'email', 'address', 'phone', 'sex', 'gender',
+            'order', 'order_id', 'orderid',
+            'payment', 'credit_card'
+        ];
+
+        // Deep copy
+        let _payload = JSON.stringify(payload);
+        attributesToMask.forEach(attr => {
+            _payload = _payload.replaceAll(new RegExp(`("${attr}"):(".+?"|\\d+)`, 'mgi'), '$1:"********"');
+        });
+
+        return _payload;
+    };
+
+    /**
+     * Send event to endpoint
+     *
+     * @param  { string } name Event name
+     * @param  { object } payload Event payload
+     * @return { object } request response
+     */
+    async function _sendEvent(name, payload) {
+        _setSessionId();
+        let url;
+
+       if (host) {
+            host = host.replaceAll(/\/+$/gm, '');
+            url = `${host}/ingest/PageViewEvent/0.0`
+        }
+
+        payload = _maskSuspiciousAttributes(payload);
+        payload = Object.assign({}, JSON.parse(payload), globalAttributes);
+        payload = JSON.stringify(payload);
+
+
+
+        const request = new XMLHttpRequest();
+        request.open('POST', url, true);
+        request.setRequestHeader('Content-Type', 'application/json');
+        request.setRequestHeader('Access-Control-Allow-Origin', '*')
+        request.send(JSON.stringify({
+            timestamp: new Date().toISOString(),
+            action: name,
+            version: '1',
+            session_id: _getSessionId(),
+            payload
+        }));
+    };
+
+    /**
+     * Track page hit
+     */
+    function _trackPageHit() {
+        _setSessionId();
+
+        // If local development environment
+        // if (/^localhost$|^127(\.[0-9]+){0,2}\.[0-9]+$|^\[::1?\]$/.test(location.hostname) || location.protocol === 'file:') return;
+        // If test environment
+        if (window.__nightmare || window.navigator.webdriver || window.Cypress) return;
+
+        let country, locale;
+        try {
+            const timezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+            country = timezones[timezone];
+            locale = (navigator.languages && navigator.languages.length) ? navigator.languages[0] : navigator.userLanguage || navigator.language || navigator.browserLanguage || 'en';
+        } catch (error) {
+            // ignore error
+        }
+
+        // Wait a bit for SPA routers
+        setTimeout(() => {
+         fetch('http://localhost:4000/ingest/PageViewEvent/0.0', {
+                method: 'POST',
+                mode: 'no-cors',
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                body: JSON.stringify(
+                    {
+                        eventId: _uuidv4(),
+                        timestamp: new Date(),
+                        version: '1',
```

Comment:
What's this version for? should it be synced with the version of the model?

Reply:
Removed since we know this from table